mypy/plugins/dataclasses.py

```diff
@@ -81,12 +91,7 @@ def transform(self) -> None:
             # Some definitions are not ready, defer() should be already called.
             return
         for attr in attributes:
-            node = info.get(attr.name)
-            if node is None:
-                # Nodes of superclass InitVars not used in __init__ cannot be reached.
-                assert attr.is_init_var and not attr.is_in_init
-                continue
-            if node.type is None:
+            if attr.type is None:
```

Comment:
What if the some attribute has no type? We seem to handle this case during deserialization, so maybe it can happen here as well. Could this result in always deferring for some attribute? If not, maybe we should require that the serialized type is not `None`?

Reply:
We only consider something as being an attribute if it has a type annotation, so the type missing is (should be) always due to resolution issues that can be fixed by deferring.

I think we can only serialize things with non-None types, yeah.